src/Chainweb/CutDB.hs

```diff
@@ -370,6 +386,32 @@ cutHashesToBlockHeaderMap headerStore payloadStore hs = do
                     return $ Left cv
                 e -> throwM e
 
+-- | Has the given `CutDb` sync'd with remote peers enough, such that we're
+-- confident that activating mining / mempool on this node wouldn't create
+-- wasted work? The threshold of "closeness" is determined from the `ChainGraph`
+-- implied by the given `ChainwebVersion`. In essence, if our current Cut is...:
+--
+--   * below the threshold: Stop mining.
+--   * above the threshold: We have joined the pack of honest miners, and can mine.
+--   * even higher than the network: We are either a "superior" fork, or we are in
+--     initial network conditions where there is no real consensus yet. In this
+--     case, we keep mining.
+--
+hasJoinedConsensus :: ChainwebVersion -> CutDb cas -> IO Bool
+hasJoinedConsensus v cutdb = readTVarIO (_cutNetworkCutHeight cutdb) >>= \case
```

Comment:
As lined out above, eventually, we should base our assessment of the current consensus on more than just the tip of the queue. For now it's fine.